src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Eabi/EabiUnwindConverter.cs

```diff
@@ -0,0 +1,280 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Numerics;
+using System.Text;
+
+using static ILCompiler.ObjectWriter.EabiNative;
+
+namespace ILCompiler.ObjectWriter
+{
+    internal static class EabiUnwindConverter
+    {
+        /// <summary>
+        /// Convert from the DWARF CFI opcodes produced by JIT into the ARM EHABI
+        /// opcodes for exception unwinding.
+        /// </summary>
+        /// <param name="blobData">DWARF CFI blob from JIT</param>
+        /// <returns>
+        /// ARM EHABI unwind code, as specified by Exception Handling ABI for the Arm
+        /// Architecture, 2023Q3, section 10.3.
+        /// </returns>
+        public static byte[] ConvertCFIToEabi(byte[] blobData)
+        {
+            if (blobData == null || blobData.Length == 0)
+            {
+                return blobData;
+            }
+
+            Debug.Assert(blobData.Length % 8 == 0);
+
+            // The maximum sequence length of the ARM EHABI unwinding code is 1024
+            // bytes.
+            byte[] unwindData = new byte[1024];
```

Comment:
```suggestion
            var unwindData = new ArrayBuilder<byte>();
```
I assume that the blob is typically much smaller. Allocating full 1024 bytes every time looks wasteful.

Reply:
Good catch. It was originally a `stackalloc` but I could not use that with the local methods. I rewrote it to use `ArrayPool` now, to match the existing code in `DwarfFde`. That said, it's probably something that can be rewritten not to allocate at all. I will likely revisit it at later time.